src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -88,6 +89,13 @@ public async Task OnConnectionAsync(ConnectionContext context)
             context.Features.Set<ITlsConnectionFeature>(feature);
             context.Features.Set<ITlsHandshakeFeature>(feature);
 
+            if (_options.HttpProtocols == HttpProtocols.Http3)
```

Comment:
MsQuic should be supplying TlsConnectionFeature so this short circuit should happen 4 lines earlier.